main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -314,35 +314,90 @@ object GenExpression {
       }
 
-    case Expr.Intrinsic0(op, tpe, loc) => op match {
+    case Expr.Intrinsic(op, exps, tpe, loc) => exps match {
 
-      case IntrinsicOperator0.Cst(cst) =>
-        compileConstant(visitor, cst, tpe, loc)
+      case Nil => op match {
 
-      case IntrinsicOperator0.Region =>
-        //!TODO: For now, just emit unit
-        compileConstant(visitor, Ast.Constant.Unit, MonoType.Unit, loc)
+        case IntrinsicOperator.Cst(cst) =>
+          compileConstant(visitor, cst, tpe, loc)
 
-      case IntrinsicOperator0.RecordEmpty =>
-        // Adding source line number for debugging
-        addSourceLine(visitor, loc)
-        // We get the JvmType of the class for the RecordEmpty
-        val classType = JvmOps.getRecordEmptyClassType()
-        // Instantiating a new object of tuple
-        visitor.visitFieldInsn(GETSTATIC, classType.name.toInternalName, BackendObjType.RecordEmpty.InstanceField.name, classType.toDescriptor)
+        case IntrinsicOperator.Region =>
+          //!TODO: For now, just emit unit
+          compileConstant(visitor, Ast.Constant.Unit, MonoType.Unit, loc)
 
-      case IntrinsicOperator0.GetStaticField(field) =>
-        addSourceLine(visitor, loc)
-        val declaration = asm.Type.getInternalName(field.getDeclaringClass)
-        visitor.visitFieldInsn(GETSTATIC, declaration, field.getName, JvmOps.getJvmType(tpe).toDescriptor)
+        case IntrinsicOperator.RecordEmpty =>
+          // Adding source line number for debugging
+          addSourceLine(visitor, loc)
+          // We get the JvmType of the class for the RecordEmpty
+          val classType = JvmOps.getRecordEmptyClassType()
+          // Instantiating a new object of tuple
+          visitor.visitFieldInsn(GETSTATIC, classType.name.toInternalName, BackendObjType.RecordEmpty.InstanceField.name, classType.toDescriptor)
 
-      case IntrinsicOperator0.HoleError(sym) =>
-        addSourceLine(visitor, loc)
-        AsmOps.compileThrowHoleError(visitor, sym.toString, loc)
+        case IntrinsicOperator.GetStaticField(field) =>
+          addSourceLine(visitor, loc)
+          val declaration = asm.Type.getInternalName(field.getDeclaringClass)
+          visitor.visitFieldInsn(GETSTATIC, declaration, field.getName, JvmOps.getJvmType(tpe).toDescriptor)
 
-      case IntrinsicOperator0.MatchError =>
-        addSourceLine(visitor, loc)
-        AsmOps.compileThrowFlixError(visitor, BackendObjType.MatchError.jvmName, loc)
+        case IntrinsicOperator.HoleError(sym) =>
+          addSourceLine(visitor, loc)
+          AsmOps.compileThrowHoleError(visitor, sym.toString, loc)
+
+        case IntrinsicOperator.MatchError =>
+          addSourceLine(visitor, loc)
+          AsmOps.compileThrowFlixError(visitor, BackendObjType.MatchError.jvmName, loc)
+
+        case _ => throw InternalCompilerException("Unexpected Intrinsic Operator for 0 Expressions", loc)
+
+      }
+
+      case e1 :: Nil => ???
+      case e1 :: e2 :: Nil => op match {
```

Comment:
The matching here seems a bit strange. I would have expected either:

```
case Expr.Intrinsic(InstrinsicOperator.Spawn, List(e1)) => ...
```

or 

```
case Expr.Intrinsic(InstrinsicOperator.Spawn, exps) => exps match
   case List(e1) => ...
   case _ => throw ...
```

but in any case, its important to match not on the order of arguments, but on the order they appear in InstrincOperator (which I would like to match the usual order of expression in e.g. FinalAst)

Reply:
Does it make sense?